src/support.rs

```diff
@@ -0,0 +1,164 @@
+        if let Err(e) = &interactor {
+            error!(
+                "Support could not initialize directory interactor: {}",
+                e.to_string()
+            );
+        }
+        let interactor = interactor.ok();
+
+        let log_dir = interactor
+            .as_ref()
+            .map(|interactor| format!("{:?}", interactor.get_directory()));
+
+        Self {
+            mailto_url: MailtoFactory::new(SUPPORT_EMAIL.to_string())
+                .with_subject("Help Needed".to_owned())
+                .with_content(EMAIL_TEMPLATE.to_owned())
+                .build(),
+            interactor,
+            log_dir,
+            most_recent_log: None,
+        }
+    }
+}
+
+static MAX_LOG_LINES: usize = 500;
+static SUPPORT_EMAIL: &str = "support@damus.io";
+static EMAIL_TEMPLATE: &str = "Describe the bug you have encountered:\n<-- your statement here -->\n\n===== Paste your log below =====\n\n";
+
+impl Support {
+    pub fn refresh(&mut self) {
+        if let Some(interactor) = &self.interactor {
+            if let Some(log_str) = get_log_str(interactor) {
+                self.most_recent_log = Some(log_str);
+            }
+        }
+    }
+
+    pub fn get_mailto_url(&self) -> &str {
+        &self.mailto_url
+    }
+
+    pub fn get_log_dir(&self) -> Option<&String> {
+        self.log_dir.as_ref()
+    }
+
+    pub fn get_most_recent_log(&self) -> Option<&String> {
+        self.most_recent_log.as_ref()
+    }
+}
+
+fn get_log_str(interactor: &FileDirectoryInteractor) -> Option<String> {
+    match interactor.get_most_recent() {
+        Ok(Some(most_recent_name)) => {
+            match interactor.get_file_last_n_lines(most_recent_name.clone(), MAX_LOG_LINES) {
+                Ok(file_contents) => return Some(file_contents),
+                Err(e) => {
+                    error!(
+                        "Error retrieving the last lines from file {}: {:?}",
+                        most_recent_name, e
+                    );
+                }
+            }
+        }
+        Ok(None) => {
+            error!("No files were found.");
+        }
+        Err(e) => {
+            error!("Error fetching the most recent file: {:?}", e);
+        }
+    }
+
+    None
+}
+
+struct MailtoFactory {
+    content: Option<String>,
+    address: String,
+    subject: Option<String>,
+    max_size: Option<usize>,
+}
+
+impl MailtoFactory {
+    fn new(address: String) -> Self {
+        Self {
+            content: None,
+            address,
+            subject: None,
+            max_size: None,
+        }
+    }
+
+    // will be truncated so the whole URL is at most 2000 characters
+    pub fn with_content(mut self, content: String) -> Self {
+        self.content = Some(content);
+        self
+    }
+
+    pub fn with_subject(mut self, subject: String) -> Self {
+        self.subject = Some(subject);
+        self
+    }
+
+    pub fn with_capacity(mut self, max_size: usize) -> Self {
```

Comment:
looks like this is never used, so we'll need to add a clippy supression. although I'm not sure what this is for.